### PR TITLE
#671 Added ability to filter jobs by status

### PR DESCRIPTION
**Pull request by dotnetrocks**

### CODE REVIEW

File: Smartling.API/Job/JobApiClient.cs [hunk 1]
```diff
@@ -78,35 +78,42 @@ public virtual AddFileToJobResponse AddLocale(string localeId, string jobId, boo
...
-        page = GetPage(jobName, PageSize, PageSize * pageNumber);
+        page = GetPage(jobName, PageSize, PageSize * pageNumber, statusFilter);
         results.AddRange(page.items);
       }
 
       return results;
     }
 
-    public virtual JobList GetPage(string jobName, int limit, int offset)
+    public virtual List<Smartling.Api.Model.Job> GetActive()
```
Comment (dimitrystd) [inline Smartling.API/Job/JobApiClient.cs]: Well, maybe it's useful to have a predefined filter in SDK but I believe every developer has their own opinion on what is "active" job. The second reason why I dislike this change this is how `ActiveJobsFilter` const looks. It takes time to figure out how it works.
Let's leave the new `Get(string jobName = "", string statusFilter = "")` and move filter string and decision to sitecore connector.

Comment (dotnetrocks) [reply Smartling.API/Job/JobApiClient.cs]: Ok, done.

File: Smartling.API/Job/JobApiClient.cs [hunk 2]
```diff
@@ -11,12 +11,12 @@ public class JobApiClient : ApiClientBase
     private readonly string CreateJobUrl = "/jobs-api/v3/projects/{0}/jobs";
     private readonly string GetJobUrl = "/jobs-api/v3/projects/{0}/jobs?limit={1}&offset={2}";
     private readonly string GetJobByIdUrl = "/jobs-api/v3/projects/{0}/jobs/{1}";
-    private readonly string GetJobByNameUrl = "/jobs-api/v3/projects/{0}/jobs?jobName={1}&limit={2}&offset={3}";
     private readonly string UpdateJobUrl = "/jobs-api/v3/projects/{0}/jobs/{1}";
     private readonly string JobFileUrl = "/jobs-api/v3/projects/{0}/jobs/{1}/file/add";
     private readonly string JobAuthorizeUrl = "/jobs-api/v3/projects/{0}/jobs/{1}/authorize";
     private readonly string AddLocaleUrl = "/jobs-api/v3/projects/{0}/jobs/{1}/locales/{2}?syncContent={3}";
     private readonly string GetProcessesUrl = "/jobs-api/v3/projects/{0}/jobs/processes";
+    private readonly string ActiveJobsFilter = "IN_PROGRESS&translationJobStatus=COMPLETED&translationJobStatus=AWAITING_AUTHORIZATION";
```
Comment (dimitrystd) [inline Smartling.API/Job/JobApiClient.cs]: Delete it

File: Smartling.API/Job/JobApiClient.cs [hunk 3]
```diff
@@ -78,35 +78,37 @@ public virtual AddFileToJobResponse AddLocale(string localeId, string jobId, boo
...
+    
+    public virtual JobList GetPage(string jobName, int limit, int offset, string statusFilter = "")
     {
-      StringBuilder uriBuilder;
-      if (string.IsNullOrEmpty(jobName))
+      var url = string.Format(GetJobUrl, projectId, limit, offset);
+      if (!string.IsNullOrEmpty(jobName))
       {
-        uriBuilder = this.GetRequestStringBuilder(string.Format(GetJobUrl, projectId, limit, offset));
+        url += "&jobName=" + jobName;
```
Comment (dimitrystd) [inline Smartling.API/Job/JobApiClient.cs]: Don't forget to urlencode all params (`jobName`)

File: Smartling.API/Job/JobApiClient.cs [hunk 4]
```diff
@@ -78,35 +78,37 @@ public virtual AddFileToJobResponse AddLocale(string localeId, string jobId, boo
...
       }
-      else
+
+      if (!string.IsNullOrEmpty(statusFilter))
       {
-        uriBuilder = this.GetRequestStringBuilder(string.Format(GetJobByNameUrl, projectId, jobName, limit, offset));
+        url += "&translationJobStatus=" + statusFilter;
```
Comment (dimitrystd) [inline Smartling.API/Job/JobApiClient.cs]: Now we can't guess what developer will pass here, so remove `translationJobStatus=`.
I worry that we can't properly urlencode `statusFilter` too.

Comment (dimitrystd) [reply Smartling.API/Job/JobApiClient.cs]: Maybe parse `statusFilter` on by key-value and concat again with encoding?

Comment (dotnetrocks) [reply Smartling.API/Job/JobApiClient.cs]: @dimitrystd Maybe pass allowed statuses as string array?

File: Smartling.API/Job/JobApiClient.cs (hunk 2 above)
Comment (dotnetrocks) [reply Smartling.API/Job/JobApiClient.cs]: Done

File: Smartling.API/Job/JobApiClient.cs (hunk 3 above)
Comment (dotnetrocks) [reply Smartling.API/Job/JobApiClient.cs]: Done

File: Smartling.API/Job/JobApiClient.cs (hunk 4 above)
Comment (dotnetrocks) [reply Smartling.API/Job/JobApiClient.cs]: @dimitrystd Please see most recent commit with example

File: Smartling.API/Job/JobApiClient.cs [hunk 5]
```diff
@@ -78,35 +78,37 @@ public virtual AddFileToJobResponse AddLocale(string localeId, string jobId, boo
...
       }
-      else
+
+      if (allowedStatuses != null)
       {
-        uriBuilder = this.GetRequestStringBuilder(string.Format(GetJobByNameUrl, projectId, jobName, limit, offset));
+        url = allowedStatuses.Aggregate(url, (current, allowedStatus) => current + $"&translationJobStatus={System.Net.WebUtility.UrlEncode(allowedStatus)}");
       }
 
+      var uriBuilder = this.GetRequestStringBuilder(url);
```
Comment (dimitrystd) [inline Smartling.API/Job/JobApiClient.cs]: It looks a little bit strange to create string builder on the last step. It seems you can easily use uriBuilder in two conditional statements above.

Comment (dotnetrocks) [reply Smartling.API/Job/JobApiClient.cs]: Updated.

**Pull request merged**